src/mono/mono/mini/mini-exceptions.c

```diff
@@ -2698,8 +2702,17 @@ mono_handle_exception_internal (MonoContext *ctx, MonoObject *obj, gboolean resu
 			}
 			g_print ("[%p:] EXCEPTION handling: %s.%s: %s\n", (void*)(gsize)mono_native_thread_id_get (), m_class_get_name_space (mono_object_class (obj)), m_class_get_name (mono_object_class (obj)), msg);
 			g_free (msg);
-			if (mono_ex && mono_trace_eval_exception (mono_object_class (mono_ex)))
+			if (mono_ex && mono_trace_eval_exception (mono_object_class (mono_ex))) {
 				mono_print_thread_dump_from_ctx (ctx);
+
+#ifdef HOST_WASM
+				EM_ASM(
+					var err = new Error();
+					console.log ("Native Stacktrace: \n");
+					console.log (err.stack);
+				);
```

Comment:
I'm not sure if we have any guidelines about Emscripten-specific stuff in the runtime like this.  I think my only hesitation would be if you were doing something that
*  needed a function from either a non-Mono-specific global JS object,
* or some local scope (ie "we know this is gonna get called from inside `library_mono.js` so we'll just refer to unqualified stuff from that scope"),
* or something that wouldn't work in a non-browser JS.

This particular usage is none of those things, so inline JS seems ok.